bitnami/grafana-alloy/templates/hpa.yaml

```
@@ -0,0 +1,42 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.autoscaling.hpa.enabled }}
```

Don't create the HPA this if `alloy.resourceType` is `daemonset`
```suggestion
{{- if and .Values.autoscaling.hpa.enabled (not (eq .Values.alloy.resourceType "daemonset")) }}
```